taichi/ir.h

```diff
@@ -1363,7 +1363,10 @@ class Block : public IRNode {
 
   template <typename T, typename... Args>
   Stmt *push_back(Args &&... args) {
-    statements.emplace_back(std::make_unique<T>(std::forward<Args>(args)...));
+    statements.emplace_back([this](std::unique_ptr<T> stmt) {
+      stmt->parent = this;
+      return stmt;
+    } (std::make_unique<T>(std::forward<Args>(args)...)));
```

Comment:
Thanks! Although this implementation may do the job, an easier and trivially correct way to do this is
```C++
auto stmt = std::make_unique<T>(std::forward<Args>(args)...);
stmt->parent = this;
statements.emplace_back(std::move(stmt));
```

If there a specific reason to use a temporary lambda here? If no please always use the easiest solutions to solve easy problems, so that you have time and energy for real hard problems :-)

Reply:
Oh, I didn't know that the `->` operator of a parameter list is also callable. Will change to the easier solution.